src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/CustomItemEffects.java

```diff
@@ -533,6 +567,44 @@ public int countItemsInInventoryAndStorage(ItemStack match) {
 		return count;
 	}
 
+	public ItemStack getFirstItemInRuler() {
```

Comment:
You already have a method for getting the data of a ruler, consider moving that shared logic into a utility method